nshlib/Kconfig

```diff
@@ -127,6 +127,8 @@ config NSH_CMDPARMS
 	bool "Enable commands as parameters"
 	default !DEFAULT_SMALL
 	depends on !DISABLE_MOUNTPOINT
+	select FS_TMPFS
+	select NSH_QUOTE
```

Comment:
should we add depends on ?

Reply:
I don't like the idea of using 'depend on' when the dependent features is not enabled by default. It will hide the current feature from user.

Reply:
Fixed, doesn't enable it

Reply:
Too bad there isn't a feature in the Kconfig menu to show all features, including those with missing dependent features. By default they would be hidden, but the user could show all features to find out what other possibilities exist. This would require opening a PR with Kconfig-frontends. I think that's [here](https://github.com/uvc-ingenieure/kconfig-frontends) now? (Not sure if that's ahead or behind the one we've used in the past.)

Alternately it would be cool if autogenerated docs from the Kconfigs could show everything and explain the dependencies. We could probably do this.

Reply:
I actually did this a year ago (closed the PR) because it took forever to render all the documentation pages because of how many configs we have :(

Reply:
> Alternately it would be cool if autogenerated docs from the Kconfigs could show everything and explain the dependencies. We could probably do this.

I actually did this a year ago (closed the PR) because it took forever to render all the documentation pages because of how many configs we have :(

Reply:
Do you have the PR still? Maybe it's possible to speed it up by using a different algorithm. If Kconfig can process it quickly, then a tool to convert it to documentation should be able to do its thing quickly, too. I'm willing to look into it if there's some kind of starting point to put me on the right path :-)

Reply:
> Do you have the PR still? Maybe it's possible to speed it up by using a different algorithm. If Kconfig can process it quickly, then a tool to convert it to documentation should be able to do its thing quickly, too. I'm willing to look into it if there's some kind of starting point to put me on the right path :-)

https://github.com/apache/nuttx/pull/3608